src/query.c

```diff
@@ -112,6 +112,9 @@ int query__batch(sqlite3_stmt *stmt, struct buffer *buffer) {
 
 	/* Insert the column count */
 	cursor = buffer__advance(buffer, sizeof(uint64_t));
+	if (cursor == NULL) {
```

Comment:
In this case I think the buffer passed here is guaranteed to have at least 8 bytes available, so perhaps this could be turned into:

```
assert(cursor != NULL);
```

but it's also fine this way.

Reply:
No, I'm fine with the assert.